### PR TITLE
Array instances using square brackets instead of parentheses

### PR DESCRIPTION
**Pull request by pieter3d**
PR to address #3092

### CODE REVIEW

**Comment by pieter3d** [comment]
Hmm - how are these error supposed to be debugged? I ran the regression as `HARNESS_UPDATE_GOLDEN=1 make test_regress` and included all changed logs with my PR

**Comment by wsnyder** [comment]
That's the right process.  Perhaps some tests were SKIPPED and so not updated because you don't have grkwave installed?

File: include/verilated_vcd_c.cpp [hunk 1]
```diff
@@ -431,9 +431,9 @@ void VerilatedVcd::dumpHeader() {
 
             for (; *np && *np != ' ' && *np != '\t'; np++) {
                 if (*np == '[') {
-                    printStr("(");
+                    printStr("[");
                 } else if (*np == ']') {
-                    printStr(")");
+                    printStr("]");
```
Comment (wsnyder) [inline include/verilated_vcd_c.cpp:436]: Suspect these code blocks can just be removed?

**Review by wsnyder — COMMENTED** [review]
One potential cleanup, then once tests pass think can be merged.

**Comment by pieter3d** [comment]
I do have gtkwave installed. The regression was run without systemC installed, so I did that. That now has a bunch of failing tests, but appears to be a build issue.
I installed systemC just by building it with cmake from the official github repo. It installs into /opt/systemc which is what I set the env var $SYSTEMC to. Anything else I should be setting? Haven't seen docs on this yet.

**Comment by pieter3d** [comment]
Errors are linking issues with undefined references to  `sc_core::` stuff.

**Comment by wsnyder** [comment]
Hmm, I saw the miscompares in the log.  Maybe refunding by pushing a change will fix it?

Locally and in github I only use the Ubuntu 20.04 SystemC, haven't tried cmake's.

**Comment by pieter3d** [comment]
Currently I am rerunning with the following env vars:
```
SYSTEMC_ARCH=linux
SYSTEMC_LIBDIR=/opt/systemc/lib
SYSTEMC_INCLUDE=/opt/systemc/include
SYSTEMC=/opt/systemc
```

Basically just filling in missing values I saw in the logs...
Perhaps that will cause the systemc tests to build/run?

File: include/verilated_vcd_c.cpp (hunk 1 above)
Comment (pieter3d) [reply include/verilated_vcd_c.cpp:436]: Actually... I tried leaving those untouched and the VCD files still used () to index signals, not [ ]. I'm not sure why though...

**Comment by gezalore** [comment]
> Any idea where those are generated?

I suspect here
https://github.com/verilator/verilator/blob/00fe36f44cd7ea2548c1eb5a52f956562e9a99da/src/V3TraceDecl.cpp#L256
and here
https://github.com/verilator/verilator/blob/00fe36f44cd7ea2548c1eb5a52f956562e9a99da/src/V3TraceDecl.cpp#L281

**Comment by pieter3d** [comment]
Yep, I found those, thanks! Currently running a local regression to confirm that addresses everything

**Comment by pieter3d** [comment]
This is ready for merge now

**Comment by wsnyder** [comment]
Thanks for your work on this.

**Pull request merged**